src/tools/kdb/main.c

```diff
@@ -135,30 +229,27 @@ int main (int argc, char ** argv)
 		const char * errorMessage = GET_ERR (parentKey);
 		if (elektraStrNCmp (errorMessage, "Unknown sub-command:", 20) == 0)
 		{
-			goto cpp;
+			result = cpp_main (argc, argv);
```

Comment:
I don't think this is the best solution, this checks all C++ commands again.

You should probably directly call the `tryExternalCommand` function from `external.cpp` (or wrapper for it), or re-implement the behaviour here (preferred). AFAICT it's basically just:

1. Load `KDB_EXEC_PATH` from environment
2. Iterate over it's entries (use `strchr(kdbExecPath, ':')` )
3. For each entry, copy it to a buffer and append `argv[1]`. Then try `execve` with the buffer. If the external file exists, `execve` won't return. If it does return, check `errno`. According to [this](https://github.com/zerovm/glibc/blob/master/posix/execvp.c), we continue with the loop on `ENOENT`, `ESTALE`, `ENOTDIR`, `ENODEV` and `ETIMEDOUT`. For all other errors, we abort and print `strerror`. (Note: I think we can ignore the special case for `EACCES`, but maybe print `strerror` in debug mode and continue anyway).
4. After the loop, try once more with `BUILTIN_EXEC_FOLDER` (`#define`d macro, not an env-var) and `argv[1]`.
5. If that also fails, report and unknown sub-command.

(Note: I don't think the `stat` in `tryExternalCommand` does anything useful, if you check `errno` after `execve`)

Reply:
#4951